### PR TITLE
[releases] utilize project global to know if releases are available

### PR DESCRIPTION
**Pull request by dcramer**
- Add Project.flags

### CODE REVIEW

**Comment by dcramer** [comment]
fyi cant use BitField in migrations, thats why im doing this raw sql/bit operation stuff

File: src/sentry/south_migrations/0297_backfill_project_has_releases.py [hunk 1]
```diff
@@ -0,0 +1,737 @@
...
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        Project = orm['sentry.Project']
+        ReleaseProject = orm['sentry.ReleaseProject']
+
+        db.commit_transaction()
+        queryset = Project.objects.all()
+        for item in RangeQuerySetWrapper(queryset):
```
Comment (mattrobenolt) [inline src/sentry/south_migrations/0297_backfill_project_has_releases.py]: No progress bar??

**Review by mattrobenolt — APPROVED** [review]
Would we want to split these into separate PRs for separate deploys?

There will be a race condition between the time the backfill runs and code is deployed, so some projects might never get their flag set.

Granted, the window is likely small, but worth noting. We could always re-run the backfill by hand to make sure we catch everything.

File: src/sentry/models/release.py [hunk 2]
```diff
@@ -125,6 +128,9 @@ def get_or_create(cls, project, version, date_added):
                         version=version
                     )
                 release.add_project(project)
+                if not project.flags.has_releases:
```
Comment (macqueen) [inline src/sentry/models/release.py]: don't you need to do this in other places that `ReleaseProject`s are created?

**Review by mattrobenolt — DISMISSED** [review]
GH-4918 won the race, so can we rename the migraton files here? The previous is only a data migration so we don't need to regenerate the frozen state.

**Comment by dcramer** [comment]
@mattrobenolt i dont really care -- future releases detected would fix it anyways so should be ok

I'm also happy to manually run the migration a second time just to push this all at once

**Pull request merged**